bouncer/shared/executable_test.ts

```diff
@@ -0,0 +1,94 @@
+import { SwapContext } from './swap_context';
+import { runWithTimeout } from './utils';
+
+export enum TestStatus {
+  Ready = 'ready',
+  Running = 'running',
+  Complete = 'complete',
+  Failed = 'failed',
+}
+
+export class ExecutableTest {
+  public status: TestStatus = TestStatus.Ready;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private runFunction: (...args: any[]) => Promise<void>;
+
+  timeoutSeconds: number;
+
+  debug = false;
+
+  public swapContext: SwapContext;
+
+  constructor(
+    public name: string,
+    testFunction: () => Promise<void>,
+    timeoutSeconds: number,
+  ) {
+    this.runFunction = testFunction;
+    this.timeoutSeconds = timeoutSeconds;
+    this.swapContext = new SwapContext();
+  }
+
+  /// Run the test with the pre-defined timeout and any given arguments
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async run(...args: any[]) {
+    if (this.status === TestStatus.Running) {
+      throw new Error(`${this.name} Test is already running`);
+    }
```

Comment:
So it can be called again after it's already completed? what's the intention with this condition specifically? Why couldn't we kick off two instances of the same test?

Reply:
Because I create the instance for the test in the files "global" context, we will only every have one instance, and we can guarantee it exists at all times. I went with this approach because:
- Way simpler than anything else i could think of. 
- Allows us to define the details of the test in one place and in the same place as the test itself.
- Allows using of the test instance for logging and other things without passing it to every function the test has.
- At the moment we never run 2 instances of a test.

If you want to still run 2 instances of a test, you can create a test that does that by using the inner function of the test instead.
```js
export const mySingleTest = new ExecutableTest('single', main, 600);
async function main(){
  // normal test 
}

export const myDoubleTest = new ExecutableTest('double', doubleTest, 600);
async function doubleTest(){
  await await Promise.all([main, main]);
}
```

Reply:
makes sense